### PR TITLE
refactor: phpcs and composer related updates

### PR DESCRIPTION
**Pull request by yigitcukuren**
Hey there,

I know this looks huge, but as far as I can see, there are no standards in the PHP code. I thought I might add some standards to make the PHP SDK more collaborative for everyone. If you think these changes are too much, I wouldn't mind creating multiple PRs for a smoother transition.

### CODE REVIEW

**Comment by bhelx** [comment]
Thanks for taking the time to improve it! We don't mind the size of the PR. This seems to be all internal changes correct? Assuming there are no external changes then i'd consider it a simple patch / cleanup. No need to break it up.

File: src/CurrentPlugin.php [hunk 1]
```diff
@@ -42,20 +42,20 @@ function read_block(int $offset) : string
 
     /**
      * Allocates a block of memory in the plugin's memory and returns the offset.
-     * 
+     *
      * @param int $size Size of the block to allocate in bytes.
      */
-    function allocate_block(int $size) : int
+    public function allocate_block(int $size): int
```
Comment (bhelx) [inline src/CurrentPlugin.php]: do these need to be public? These are fairly low-level methods that we don't want people calling in normal operation.

File: src/LibExtism.php [hunk 2]
```diff
@@ -38,41 +39,42 @@ function findSo(string $name): \FFI {
...
     }
 
-    function extism_current_plugin_memory(FFI\CData $plugin): \FFI\CData
+    public function extism_current_plugin_memory(FFI\CData $plugin): \FFI\CData
     {
         return $this->ffi->extism_current_plugin_memory($plugin);
     }
 
-    function extism_current_plugin_memory_free(FFI\CData $plugin, FFI\CData $ptr): void
+    public function extism_current_plugin_memory_free(FFI\CData $plugin, FFI\CData $ptr): void
```
Comment (bhelx) [inline src/LibExtism.php]: does this expose the methods to the external users? This is an internal API.

Comment (yigitcukuren) [reply src/LibExtism.php]: They were already public if you not defined them explicitly protected or private.

From php.net
```
Class methods may be defined as public, private, or protected. 
Methods declared without any explicit visibility keyword are defined as public.
```

File: src/CurrentPlugin.php (hunk 1 above)
Comment (yigitcukuren) [reply src/CurrentPlugin.php]: Actually I am not completely sure what does these function behind the scene. As far as I checked it is used in HostFunctions.php. If you call these methods in the HostFunctions they should be public, if you don't you can set them as private.

Comment (yigitcukuren) [reply src/CurrentPlugin.php]: I've pushed some code and I changed the visibility of unused functions in CurrentPlugin.php. You can guide me about updating visibility of specific methods.

File: src/LibExtism.php (hunk 2 above)
Comment (yigitcukuren) [reply src/LibExtism.php]: Since this function called in CurrentPlugin, we can't make it private.

**Review by mhmd-azeez — CHANGES_REQUESTED** [review]
Great job @yigitcukuren, thank you very much! Can you please go through LibExtism.php functions and try to mark as many of them private as possible?

**Comment by yigitcukuren** [comment]
> Great job @yigitcukuren, thank you very much! Can you please go through LibExtism.php functions and try to mark as many of them private as possible?

Thank you! I have just updated the visibility of all the internal methods.

File: src/LibExtism.php (hunk 2 above)
Comment (bhelx) [reply src/LibExtism.php]: okay, just making sure we expose as little API to the public as possible. Maybe the class should be less accessible? I don't know if there is an equivalent of package only. But I'll leave @mhmd-azeez to pick out any changes that are needed. Thanks so much for this cleanup!

Comment (mhmd-azeez) [reply src/LibExtism.php]: Yeah, it seems like there is no package level access modifier in php. As a middle ground, I moved `LibExtism.php` to `\Extism\Internal` namespace, which is a pattern used by some php libraries like Symfony

**Review by mhmd-azeez — APPROVED** [review]
Thank you very much for your excellent work @yigitcukuren, I moved LibExtism to `\Extism\Internal` namespace to make it clear that this class is for internal use only. Let me know if you have any concerns or issues with that. Otherwise, I can go on and merge the PR

**Comment by yigitcukuren** [comment]
> Thank you very much for your excellent work @yigitcukuren, I moved LibExtism to `\Extism\Internal` namespace to make it clear that this class is for internal use only. Let me know if you have any concerns or issues with that. Otherwise, I can go on and merge the PR

It is perfectly fine! Please go ahead 🙏

**Pull request merged**